### PR TITLE
Bugfix/assembler memory

### PR DESCRIPTION
**Pull request by SandyRogers**
## PR checklist

This PR improves the retry-with-more-memory logic, by setting `maxRetries` directive to a parameter (default 5) specifically for the spades/megahit assembly modules. The more-memory-with-each-attempt was already present, this just allows those retries to happen.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/ebi-metagenomics/miassembler/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -208,6 +208,22 @@
...
+                    "description": "Maximum number of task attempt retries for (meta)spades assembly steps only.",
+                    "default": 5,
+                    "fa_icon": "fas fa-repeat",
+                    "hidden": true,
+                    "help_text": "Each retry will increase the memory by 50%. Use to limit how many times this increase-and-retry happens."
+                },
+                "max_megahit_retries": {
+                    "type": "integer",
+                    "description": "Maximum number of task attempt retries for megahit assembly steps only.",
+                    "default": 5,
```
Comment (mberacochea) [inline nextflow_schema.json:223]: 5 it's probably too much

File: nextflow.config [hunk 2]
```diff
@@ -89,6 +89,8 @@ params {
     max_memory                       = '1.TB'
     max_cpus                         = 32
     max_time                         = '168.h' // 7 days
+    max_spades_retries               = 5
```
Comment (mberacochea) [inline nextflow.config:92]: I would set this to 3, 5 could cause the whole job to take more than week in some cases (which slurm won't like)

**Review by mberacochea — CHANGES_REQUESTED** [review]
Very cool :+1:

**Review by mberacochea — APPROVED** [review]
:shipit:

**Review by KateSakharova — APPROVED** [review]
perfection!

**Pull request merged**